repl/evaluator.go

```diff
@@ -174,29 +202,37 @@ func (e *Evaluator) DelLetVar(name string) error {
 }
 
 func (e *Evaluator) applyContext() (*cel.Env, interpreter.Activation, error) {
-	env := e.env
 	var vars = make(map[string]interface{})
 	for _, el := range e.ctx.letVars {
+		if el.prog == nil {
+			continue
+		}
+
 		val, _, err := (*el.prog).Eval(vars)
 		if val != nil {
 			vars[el.identifier] = val
 		} else if err != nil {
 			return nil, nil, err
 		}
-
-		env = el.env
 	}
 
 	act, err := interpreter.NewActivation(vars)
 	if err != nil {
 		return nil, nil, err
 	}
 
+	var env *cel.Env
```

Comment:
The common idiom within go is to assign the default and then override:

```go
env := e.env
if len(e.ctx.letVars) > 0 {
  env = e.ctx.letVars[len(e.ctx.letVars)-1].env
}
```

Reply:
Done